client/source/js/modules/ui/type-selector/type-selector-ctrl.js

```diff
@@ -1,9 +1,10 @@
 define(['./module', 'underscore'], function (module, _) {
```

Comment:
Missing main comment.

For every controller (or directive) it would be good practice to add on top its main comment stating *the reason for being* of that thing.